pkg/isoeditor/rhcos.go

```diff
@@ -28,44 +28,63 @@ func NewEditor(dataDir string) Editor {
 	return &rhcosEditor{workDir: dataDir}
 }
 
-// CreateMinimalISOTemplate Creates the template minimal iso by removing the rootfs and adding the url
-func (e *rhcosEditor) CreateMinimalISOTemplate(fullISOPath, rootFSURL, arch, minimalISOPath string) error {
-	extractDir, err := os.MkdirTemp(e.workDir, "isoutil")
-	if err != nil {
-		return err
-	}
-
-	if err = Extract(fullISOPath, extractDir); err != nil {
-		return err
-	}
-
-	if err = os.Remove(filepath.Join(extractDir, "images/pxeboot/rootfs.img")); err != nil {
+func createMinimalISOFromDir(minimalISOPath, extractDir, volumeID, rootFSURL, arch string) error {
+	if err := os.Remove(filepath.Join(extractDir, "images/pxeboot/rootfs.img")); err != nil {
 		return err
 	}
 
-	if err = embedInitrdPlaceholders(extractDir); err != nil {
+	if err := embedInitrdPlaceholders(extractDir); err != nil {
 		log.WithError(err).Warnf("Failed to embed initrd placeholders")
 		return err
 	}
 
-	if err = fixGrubConfig(rootFSURL, extractDir); err != nil {
+	if err := fixGrubConfig(rootFSURL, extractDir); err != nil {
 		log.WithError(err).Warnf("Failed to edit grub config")
 		return err
 	}
+
 	// ignore isolinux.cfg for ppc64le because it doesn't exist
 	if arch != "ppc64le" {
-		if err = fixIsolinuxConfig(rootFSURL, extractDir); err != nil {
+		if err := fixIsolinuxConfig(rootFSURL, extractDir); err != nil {
 			log.WithError(err).Warnf("Failed to edit isolinux config")
 			return err
 		}
 	}
 
+	if err := Create(minimalISOPath, extractDir, volumeID); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateMinimalISOTemplate Creates the template minimal iso by removing the rootfs and adding the url
+func (e *rhcosEditor) CreateMinimalISOTemplate(fullISOPath, rootFSURL, arch, minimalISOPath string) error {
+	extractDir, err := os.MkdirTemp(e.workDir, "isoutil")
+	if err != nil {
+		return err
+	}
+
+	if err = Extract(fullISOPath, extractDir); err != nil {
+		return err
+	}
+
 	volumeID, err := VolumeIdentifier(fullISOPath)
 	if err != nil {
 		return err
 	}
 
-	if err = Create(minimalISOPath, extractDir, volumeID); err != nil {
+	err = createMinimalISOFromDir(minimalISOPath, extractDir, volumeID, rootFSURL, arch)
```

Comment:
nit: the change of argument order here, where `minimalISOPath` went from the end to the beginning (in place of `fullISOPath`) makes for an API that is prone to mistakes/misunderstandings.

Reply:
Done